src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -141,13 +145,18 @@ def _execute_sql_script(self, sql_script_path):
         This assumes that a relevant warehouse is already active.
         Consequently, "use database" will be executed first if it is set in definition file or in the current connection.
         """
-        with open(sql_script_path) as f:
-            sql_script = f.read()
+        env = get_sql_cli_jinja_env(
```

Comment:
We exploded snowflake_sql_jinja_render, any reason we couldn't modify the existing abstraction instead?

Reply:
Also, we're diverging between how package scripts are executed and how post-deploy scripts are executed. For package scripts, the env is created differently, and AFAICT if a script fails to expand we guarantee that no script is executed. This doesn't appear to be the case here.

Reply:
I have refactored to reuse some of the existing package script logic, and also make sure that we check for template errors before executing any of the scripts.